src/Microsoft.ML.Mkl.Components/SymSgdClassificationTrainer.cs

```diff
@@ -664,6 +664,10 @@ private TPredictor TrainCore(IChannel ch, RoleMappedData data, LinearModelParame
             int numFeatures = data.Schema.Feature.Value.Type.GetVectorSize();
             var cursorFactory = new FloatLabelCursor.Factory(data, CursOpt.Label | CursOpt.Features);
             int numThreads = 1;
+
+            if (_options.NumberOfThreads.HasValue)
+                numThreads = _options.NumberOfThreads.Value;
```

Comment:
>numThreads [](start = 16, length = 10)

So we now can have multithreading for SymSGD always?
If yes:
```
  /// <summary>
            /// Degree of lock-free parallelism. Determinism not guaranteed if this is set to higher than 1.
            /// Multi-threading is not supported currently.
            /// </summary>
            [Argument(ArgumentType.AtMostOnce, HelpText = "Degree of lock-free parallelism. Determinism not guaranteed. " +
                "Multi-threading is not supported currently.", ShortName = "nt")]
            public int? NumberOfThreads;
```
this need some polishing in description, but that minor.

So if it's null (which it is by default) it's would be shame to utilize only one thread.

`numThreads = NumberOfThreads ?? Environment.ProcessorCount` maybe?

Reply:
Thanks for bringing this up - I had similar thoughts and actually meant to bring this up in the PR. I agree - - null should default to the number of threads that are available to openmp. Im happy to update this...